src/main/java/saopig/command/ListCommand.java

```diff
@@ -0,0 +1,162 @@
+package saopig.command;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+
+import saopig.SaopigInvaildSizeException;
+import saopig.Storage;
+import saopig.Ui;
+import saopig.task.Deadline;
+import saopig.task.Event;
+import saopig.task.Task;
+import saopig.task.TaskList;
+
+
+/**
+ * Represents a command to list tasks.
+ */
+public class ListCommand extends Command {
+    private static final String TIME_PATTERN = "yyyy-MM-dd HH:mm";
+    private static final DateTimeFormatter DATE_TIME_FORMATTER = DateTimeFormatter.ofPattern(TIME_PATTERN);
+    private static final String FILE_PATH = "./data/saopigTaskList.txt";
+    private static final String FILE_DIRECTORY = "./data";
+
+    private String command;
+    private int typeIndex; //0 for list all, 1 for listtaskondate
+
+    /**
+     * Constructs a list command.
+     *
+     * @param command   The command.
+     * @param typeIndex The type index.
+     */
+    public ListCommand(String command, int typeIndex) {
+        this.command = command;
+        this.typeIndex = typeIndex;
+    }
+
+    private static void checkValue(int value, int lowerBound, int upperBound) throws SaopigInvaildSizeException {
+        if (value < lowerBound || value > upperBound) {
+            throw new SaopigInvaildSizeException("Error");
+        }
+    }
+
+    /**
+     * Lists all tasks.
+     * If there are no tasks, prints a message.
+     *
+     * @param taskList The task list.
+     * @param ui       The user interface.
+     */
+    public String listTasks(TaskList taskList, Ui ui) {
+        if (taskList.getTasks().isEmpty()) {
+            return ("\n"
+                    + "Oh dear, it looks like there are no tasks yet!\n "
+                    + "But that's alright.\n "
+                    + "It gives us a chance to start fresh and dream up some new plans.\n "
+                    + "Whenever you're ready to add tasks, I'll be right here to assist you.\n "
+                    + "Let's make it a magical journey together!");
+        }
+        StringBuilder response = new StringBuilder();
+        for (int i = 0; i < taskList.getTasks().size(); i++) {
+            Task task = taskList.getTasks().get(i);
+            response.append((i + 1)).append(". ").append(task.toString());
+        }
+        return response.toString();
+    }
+
+    /**
+     * Lists tasks on a date.
+     * If there are no tasks on the date, prints a message.
+     * If the date is invalid, prints a message.
+     * If the command is invalid, prints a message.
+     * If the command is incomplete, prints a message.
+     * If the command is empty, prints a message.
+     *
+     * @param input    The input.
+     * @param taskList The task list.
+     * @param ui       The user interface.
+     * @return response to the user.
```

Comment:
I think the first word after return should start with a capital letter? Maybe you can clarify and make some changes since it's a bit inconsistent in the whole codebase.